YakimS/assignment1/Q2

```diff
@@ -0,0 +1,89 @@
+public class Assignment1Q2 {
+
+    public static void main(String[] args) {
+        MyLinkedList<Integer> myLinkedList = new MyLinkedList<>();
+        Integer[] ob = new Integer[10];
+        for(int i = 0 ; i < ob.length ; i++)
+        {
+            ob[i] = i;
+            myLinkedList.add(ob[i]);
+            System.out.print(ob[i] + "");
+        }
+        System.out.println("");
+        System.out.print("kth ele:"+myLinkedList.getKthLastElement(9));
+    }
+
+
+    public static class MyLinkedList<T>
+    {
+        private MyNode<T> head;
+
+        public MyNode<T> getHead() {
+            return head;
+        }
+
+        public boolean isEmpty()
+        {
+            return head==null;
+        }
+
+        public void add(T data)
+        {
+            if (isEmpty()) head = new MyNode<>(data);
+            else
+            {
+                MyNode lastNode = head;
+                while (lastNode.getNext() != null) lastNode = lastNode.getNext();
+                lastNode.setNext(new MyNode<>(data));
+            }
+        }
+
+        public T getKthLastElement(int k)
+        {
+            MyNode currentNode = getHead();
+            for(int i = 0 ; i < k ; i++)
+            {
+                if (currentNode == null) return null;
+                currentNode = currentNode.getNext();
+            }
+
+            MyNode kthBackNode = getHead();
+            while (currentNode.getNext() != null)
+            {
+                kthBackNode = kthBackNode.getNext();
+                currentNode = currentNode.getNext();
+            }
+
+            return (T) kthBackNode.getData();
+
+        }
+
+    }
+
+    public static class MyNode<T>
```

Comment:
`MyNode` is not used (and should not be used) anywhere outside `MyLinkedList`, so it would be better to make it a private inner class of `MyLinkedList`. Could you change code to do that?